app/controllers/api/v1/visualization/clusters_controller.rb

```diff
@@ -146,11 +152,29 @@ def self.get_cluster_viz_data(study, cluster, url_params)
             return nil
           end
 
-          subsample = url_params[:subsample].blank? ? nil : url_params[:subsample]
+          subsample = url_params[:subsample].blank? ? nil : url_params[:subsample].to_i
 
           colorscale = url_params[:colorscale].blank? ? 'Reds' : url_params[:colorscale]
 
-          coordinates = ClusterVizService.load_cluster_group_data_array_points(study, cluster, annotation, subsample, colorscale)
+          is_annotated_scatter = !url_params[:is_annotated_scatter].blank?
+
+          gene_name = url_params[:gene]
+          if gene_name.blank?
+            # For "Clusters" tab in default view of Explore tab
+            coordinates = ClusterVizService.load_cluster_group_data_array_points(study, cluster, annotation, subsample, colorscale)
+          else
+            # For single-gene view of Explore tab
+            gene = study.genes.by_name_or_id(gene_name, study.expression_matrix_files.map(&:id))
+            title = ExpressionVizService.load_expression_axis_title(study)
+            if is_annotated_scatter
+              # For "Annotated scatter" tab, shown in first tab for numeric annotations
+              coordinates = ExpressionVizService.load_annotation_based_data_array_scatter(study, gene, cluster, annotation, subsample, title)
```

Comment:
is there a reason we don't pass in the color scale here?

Reply:
It's omitted because the front-end does not support changing color scale for "Annotated scatter" plots.  (The "Scatter" view option only applies to non-annotated scatter plots for numeric annotations.)